### PR TITLE
Feature/tender

### PR DESCRIPTION
**Pull request by tamaragovorcin**

### CODE REVIEW

File: IntegrationWithPharmacies/FileProtocol/ReportText.cs [hunk 1]
```diff
@@ -44,32 +44,33 @@ public String getReportText(DateOfOrder date)
...
-                stringBuilder.Append(getText(date, order, stringBuilder));
-                totalQuatity += getQuantity(date, totalQuatity, order);
+                stringBuilder.Append(getText(date, tender));
+                totalQuatity += getQuantity(date, tender);
             }
             return stringBuilder.Append("\n\n   Total ordered quatity: " + totalQuatity + "\n").ToString();
         }
 
-        public String getText(DateOfOrder date, DoctorsOrder order, StringBuilder stringBuilder)
+        public String getText(DateOfOrder date, Tender tender)
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/ReportText.cs:55]: Forward stringBuilder as a method parameter.

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 2]
```diff
@@ -10,16 +13,65 @@ public class SmptServerService
...
+        }
+
+        private static StringBuilder BasicEmailTenderInformation(string pharmacyApi)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.Append("Winner of tender is pharmacy with api key : " + pharmacyApi);
+            stringBuilder.Append(Environment.NewLine);
+            stringBuilder.Append("Requested medicines:    Name  /  Quantity  /  Price");
+            stringBuilder.Append(Environment.NewLine);
+            return stringBuilder;
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:50]: Write this in one line of code.

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 3]
```diff
@@ -10,16 +13,65 @@ public class SmptServerService
...
+        }
+
+        private static StringBuilder InformationAboutEveryMedicineInTender(List<MedicineTenderOffer> medicinesWithQuantity, StringBuilder stringBuilder)
+        {
+            foreach (MedicineTenderOffer medicineTenderOffer in medicinesWithQuantity)
+            {
+                stringBuilder.Append(medicineTenderOffer.MedicineName + "  /  " + medicineTenderOffer.AvailableQuantity + "  /  " + medicineTenderOffer.Price );
+                stringBuilder.Append(Environment.NewLine);
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:58]: Write this in one line of code.

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineWithQuantityService.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        private void UpdateOneMedicineQuantity(MedicineWithQuantity medicineWithQuantity)
+        {
+            MedicineWithQuantity medicine = GetAll().SingleOrDefault(medicineName => medicineName.Name == medicineWithQuantity.Name);
+            if (medicine != null) MedicineWithQuantityRepository.UpdateQuantity(medicine, medicineWithQuantity.Quantity);
+            else { Create(MedicineWithQuantityAdapter.MedicineWithQuantityToMedicineWithQuantityDto(medicineWithQuantity)); }
+        }
+
+        public MedicineWithQuantity createIMedicineDescription(MedicineWithQuantityDto dto)
+        {
+            MedicineWithQuantity MedicineWithQuantity = IMedicineWithQuantityRepository.GetAll().SingleOrDefault(medicine => CheckMedicineNameEquality(dto, medicine));
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineWithQuantityService.cs:74]: Change name of variable: MedicineWithQuantity->medicineWithQuantity

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineWithQuantityService.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            MedicineWithQuantity medicine = medicineWithQuantities.SingleOrDefault(medicineName => medicineName.Name == medicineTenderOffer.MedicineName);
+            if (medicine != null)  UpdateMedicine(medicineTenderOffer, medicine); 
+            else{ CreateNewMedicineWithQuantity(medicineTenderOffer);  }
+        }
+
+        public string GetMedicineDescriptionFromDatabase(string medicine)
+        {
+            MedicineWithQuantity medicineWithQuantity = GetAll().SingleOrDefault(medicineName => medicineName.Name.Equals(medicine));
+            if (medicineWithQuantity == null || medicineWithQuantity.Description.Equals("")) return null;
+            return medicineWithQuantity.Description;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineWithQuantityService.cs:94]: Write in the form of a ternary operator.

**Review by trki293 — CHANGES_REQUESTED** [review]
Modify the indicated places in the code.

**Review by trki293 — APPROVED** [review]
PR is approved. Good principles of Clean code are followed.

**Review by Maja0505 — APPROVED** [review]
PR is approved. Good Clean code practice are followed.

**Pull request merged**